clippy_lints/src/collapsible_if.rs

```diff
@@ -163,6 +164,54 @@ impl CollapsibleIf {
         }
     }
 
+    fn check_collapsible_if_nested_if_else(context: &LateContext<'_>, if_expr: &Expr<'_>) {
+        let ExprKind::If(cond, then, else_opt) = if_expr.kind else {
+            return;
+        };
+
+        let ExprKind::Block(then_block, _) = then.kind else {
+            return;
+        };
+
+        let Some(then_expr) = then_block.expr else {
+            return;
+        };
+
+        let ExprKind::If(inner_cond, inner_then, _) = then_expr.kind else {
+            return;
+        };
+
+        let Some(else_expr) = else_opt else {
+            return;
+        };
+
+        let mut spanless_eq = SpanlessEq::new(context);
+
+        if !spanless_eq.eq_expr(inner_then, else_expr) {
+            return;
+        }
+
+        span_lint_and_then(
+            context,
+            COLLAPSIBLE_IF,
+            if_expr.span,
+            "this `if` statement can be collapsed.",
+            |diag| {
+                diag.span_suggestion(
+                    if_expr.span,
+                    "collapse else and nested if blocks",
+                    format!(
+                        "if !{} || {} {}",
```

Comment:
Please add a test case with the first condition being a binary operation such as `s != "foobar"`: in this case, the `!` will be applied to `s` which is wrong.

You should use the `Sugg` interface to properly invert conditions and get parentheses if needed.